crates/matrix-sdk/tests/integration/client.rs

```diff
@@ -1490,3 +1490,53 @@ async fn test_room_sync_state_after() {
     let member = room.get_member_no_sync(user_id!("@invited:localhost")).await.unwrap().unwrap();
     assert_eq!(*member.membership(), MembershipState::Leave);
 }
+
+#[async_test]
+async fn test_server_version() {
+    use matrix_sdk::test_utils::mocks::MatrixMockServer;
+    use serde_json::json;
+    use wiremock::{matchers::{method, path}, Mock, ResponseTemplate};
+
+    let server = MatrixMockServer::new().await;
+    let client = server.client_builder().build().await;
+
+    // Mock the federation version endpoint
+    Mock::given(method("GET"))
+        .and(path("/_matrix/federation/v1/version"))
+        .respond_with(ResponseTemplate::new(200).set_body_json(json!({
+            "server": {
+                "name": "Synapse",
+                "version": "1.70.0"
+            }
+        })))
+        .mount(server.server())
+        .await;
```

Comment:
Can you add a new method in `MatrixMockServer` that would return a new endpoint, that's configurable with an `ok(&self, server_name: &str, version: &str)` method to indicate a mock a successful response?

Then you can use it in this test.